src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -828,26 +844,25 @@ public ProjectState RemoveAllNormalDocuments()
     }
 
     public ProjectState UpdateDocument(DocumentState newDocument, bool contentChanged)
-        => UpdateDocuments([newDocument], contentChanged);
+        => UpdateDocuments([DocumentStates.GetRequiredState(newDocument.Id)], [newDocument], contentChanged);
 
-    public ProjectState UpdateDocuments(ImmutableArray<DocumentState> newDocuments, bool contentChanged)
+    public ProjectState UpdateDocuments(ImmutableArray<DocumentState> oldDocuments, ImmutableArray<DocumentState> newDocuments, bool contentChanged)
     {
-        var oldDocuments = newDocuments.SelectAsArray(d => DocumentStates.GetRequiredState(d.Id));
-        if (oldDocuments.SequenceEqual(newDocuments))
-            return this;
-
-        // Must not be empty as we would have otherwise bailed out in the check above.
-        Contract.ThrowIfTrue(newDocuments.IsEmpty);
+        Contract.ThrowIfTrue(oldDocuments.IsEmpty);
```

Comment:
>Contract.ThrowIfTrue(oldDocuments.IsEmpty); [](http://example.com/codeflow?start=8&length=43)

Just wanted to make sure this is a new requirement of the callers

Reply:
Not sure I follow. The contract has been there before.

Reply:
But it would have returned before hitting the contract failure previously, right?